roles/openshift_storage_glusterfs/tasks/main.yml

```diff
@@ -0,0 +1,192 @@
+---
+- name: Create temp directory for doing work in
+  command: mktemp -d /tmp/openshift-glusterfs-ansible-XXXXXX
+  register: mktemp
+  changed_when: False
+  check_mode: no
+
+- name: Copy the admin client config(s)
+  command: >
+    cp {{ openshift.common.config_base }}/master/admin.kubeconfig {{ mktemp.stdout }}/admin.kubeconfig
+  changed_when: False
+  check_mode: no
+
+- name: Verify target namespace exists
+  oc_project:
+    kubeconfig: "{{ mktemp.stdout }}/admin.kubeconfig"
+    state: present
+    name: "{{ openshift_storage_glusterfs_namespace }}"
+  when: openshift_storage_glusterfs_is_native or openshift_storage_glusterfs_heketi_is_native
+
+- include: glusterfs_deploy.yml
+  when: openshift_storage_glusterfs_is_native
+
+- name: Make sure heketi-client is installed
+  package: name=heketi-client state=present
+
+- name: Delete pre-existing heketi resources
```

Comment:
Kind of. This is to account for trying to re-run the installer if something has gone wrong with the heketi deployment, in which case the main thing to be done is to destroy and recreate the resources. This is part of the attempt to make this playbook vaguely idempotent. The variable `openshift_storage_glusterfs_heketi_wipe` in `when` is False by default, so this task is not normally run unless you know what you're doing.